lib/models/tic_tac_toe_game.rb

```diff
@@ -35,29 +35,40 @@ class GameModelError < StandardError
     # board state nxn array of integers. 0 neutral. 1 crosses. -1 circles.
     # board status 'pending' 'in_play' 'crosses' (win) 'circles' (win) 'draw'
     class TicTacGame
-
+      attr_accessor :state
       def initialize(board=nil, current_player=1, state=:pending)
-        @board = board || EMPTY_BOARD
+        board_to_load=board || EMPTY_BOARD
+        #gotta deep clone the state
+        @board =JSON.load(JSON.dump(board_to_load))
         @current_player = current_player
         @state=state
       end
-
       attr_reader :board, :state, :current_player
+      def move(player,move)
+        if player == 0
+          player=-1
+        elsif player == 1
+          player=1
```

Comment:
I think this thing is at an early enough stage we can settle on a single format for the players and leave it like that, rather than transforming them in the #move method

Reply:
OK will do

Reply:
It'll have to be 0..n